core/src/main/java/com/google/zxing/oned/Code39Writer.java

```diff
@@ -86,4 +96,74 @@ private static void toIntArray(int a, int[] toReturn) {
     }
   }
 
+  private static String tryToConvertToExtendedMode(String contents) throws IllegalArgumentException {
+     int length = contents.length();
+     StringBuilder extendedContent = new StringBuilder();
+     for (int i = 0; i < length; i++) {
+       int character = (int)contents.charAt(i);
```

Comment:
Nit: space after cast. But can you not switch on a char? It might be more descriptive because the case labels would be chars